src/kemal/param_parser.cr

```diff
@@ -57,6 +62,23 @@ module Kemal
       end
     end
 
+    def parse_file_upload
+      HTTP::FormData.parse(@request) do |field, data, meta, headers|
+        if !meta.nil?
+          filename = meta.filename
+          if !filename.nil?
+            tempfile = Tempfile.new(filename)
+            ::File.open(tempfile.path, "w") do |file|
+              IO.copy(data, file)
+            end
+            @files[field] = FileUpload.new(tmpfile: tempfile, tmpfile_path: tempfile.path, meta: meta, headers: headers)
+          else
+            @body[field] = data.gets_to_end
```

Comment:
Assuming that kemal is run behind nginx - even if it's highly recommended - will lead to a codebase where you HAVE to run it behind nginx. It's best not to assume that now, so that it doesn't bite you later.